### PR TITLE
Suppress `ExternalLoginUserNotExist` error

### LINKED ISSUE #21202: UpdateExternalUser failed: external login user link does not exists
### Description

I just updated our staging environment from 1.16.9 to 1.17.2. Everything seems fine, but I receive 
`...rs/web/auth/oauth.go:1045:handleOAuth2SignIn() [E] [6326e27a-2] UpdateExternalUser failed: external login user link does not exists [userID: 3, loginSourceID: 1]` in the logs whenever I login via OAuth2. The login itself works as expected, but I fear this may cause issues down the road. 
My OAuth2 provider is OpenID Connect via Keycloak.
`gitea doctor --all` says everythings fine. 

The debug log for such a sign-in looks like this:
```
2022/09/18 11:18:46 ...s/context/context.go:219:HTML() [D] [6326e276] Template: user/auth/signin
2022/09/18 11:18:46 [6326e276] router: completed GET /user/login?redirect_to=%2f for <IP:PORT>, 200 OK in 2.6ms @ auth/auth.go:149(auth.SignIn)
2022/09/18 11:18:50 [6326e27a] router: completed GET /user/oauth2/keycloak for <IP:PORT>, 307 Temporary Redirect in 2.0ms @ auth/oauth.go:775(auth.SignInOAuth)
2022/09/18 11:18:50 ...rs/web/auth/oauth.go:1045:handleOAuth2SignIn() [E] [6326e27a-2] UpdateExternalUser failed: external login user link does not exists [userID: 3, loginSourceID: 1]
2022/09/18 11:18:50 [6326e27a-2] router: completed GET /user/oauth2/keycloak/callback?state=...&session_state=...&code=... for <IP:PORT>, 303 See Other in 15.1ms @ auth/oauth.go:809(auth.SignInOAuthCallback)
2022/09/18 11:18:50 ...s/context/context.go:219:HTML() [D] [6326e27a-3] Template: user/dashboard/dashboard
2022/09/18 11:18:50 [6326e27a-3] router: completed GET / for <IP:PORT>, 200 OK in 4.3ms @ web/home.go:26(web.Home)
2022/09/18 11:18:50 [6326e27a-4] router: completed GET /repo/search?count_only=1&uid=3&team_id=0&q=&page=1&mode= for <IP:PORT>, 0  in 2.7ms @ repo/repo.go:468(repo.SearchRepo)
2022/09/18 11:18:50 [6326e27a-6] router: completed GET /repo/search?sort=updated&order=desc&uid=3&team_id=0&q=&page=1&limit=15&mode=&archived=false for <IP2:PORT>, 200 OK in 1.8ms @ repo/repo.go:468(repo.SearchRepo)
2022/09/18 11:18:52 [6326e27c] router: completed GET / for <IP2:PORT>, 200 OK in 1.1ms @ web/home.go:26(web.Home)
2022/09/18 11:18:52 ...s/context/context.go:219:HTML() [D] [6326e27c] Template: home
```

### Gitea Version

1.17.2

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

I am running Gitea as docker image `gitea/gitea:1.17.2`. The DB is `mariadb:10.9.2`.

### Database

MySQL

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #21202
Closes #21276

An `ExternalLoginUser` is not mandatory if the current user account was created with/by the external login source.

### CODE REVIEW

File: routers/web/auth/auth.go [hunk 1]
```diff
@@ -619,7 +619,9 @@ func handleUserCreated(ctx *context.Context, u *user_model.User, gothUser *goth.
 	// update external user information
 	if gothUser != nil {
 		if err := externalaccount.UpdateExternalUser(u, *gothUser); err != nil {
-			log.Error("UpdateExternalUser failed: %v", err)
+			if !user_model.IsErrExternalLoginUserNotExist(err) {
```
Comment (delvh) [inline routers/web/auth/auth.go]: Can't we use `errors.Is(err, util.ErrNotExist)` now?
```suggestion
			if !errors.Is(err, util.ErrNotExist) {
```

File: routers/web/auth/oauth.go [hunk 2]
```diff
@@ -1068,7 +1068,9 @@ func handleOAuth2SignIn(ctx *context.Context, source *auth.Source, u *user_model
 
 		// update external user information
 		if err := externalaccount.UpdateExternalUser(u, gothUser); err != nil {
-			log.Error("UpdateExternalUser failed: %v", err)
+			if !user_model.IsErrExternalLoginUserNotExist(err) {
```
Comment (delvh) [inline routers/web/auth/oauth.go]:
```suggestion
			if !errors.Is(err, util.ErrNotExist) {
```

**Review by 6543 — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Comment by lunny** [comment]
CI failure is related.

**Comment by zeripath** [comment]
Ha! is this the first PR to use the new not exist err?

**Review by zeripath — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
I guess we should try to fix these missing these missing ExternalLoginUser entries though.

**Comment by KN4CK3R** [comment]
> I guess we should try to fix these missing these missing ExternalLoginUser entries though.

But from my understanding they are not missing. See my comment [here](https://github.com/go-gitea/gitea/pull/21276#issuecomment-1283003218).

**Comment by lunny** [comment]
Please send backport

**Pull request merged**